app/View/Components/SelectEdicion.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace App\View\Components;
+
+use Closure;
+use Illuminate\Contracts\View\View;
+use Illuminate\View\Component;
+use App\Models\Edicion;
+
+class SelectEdicion extends Component
+{
+    public $ediciones;
+    /**
+     * Create a new component instance.
+     */
+    public function __construct()
+    {
+        $this->ediciones = Edicion::all();
```

Comment:
Normalmente, la inicialización de las propiedades del componente utilizando el constructor se utiliza para cuando se le [pasan datos al compoente](https://laravel.com/docs/10.x/blade#passing-data-to-components). En este caso, los datos se obtienen directamente desde la base de datos, por lo que se podrían pasar directamente a la vista:

```php
    public function render(): View|Closure|string
    {
        return view('components.select-edicion', [
          'ediciones' => Edicion::all(),
        ]);
    }
```